roles/roles.go

```diff
@@ -0,0 +1,58 @@
+package roles
+
+import (
+	"slices"
+
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/services"
+)
+
+// Role is the string literal of a role.
+type Role string
+
+// RoleRegistration is a struct that contains the display name and roles for a service.
+type RoleRegistration struct {
```

Comment:
Should we just call this 
```suggestion
type EntitleMetadata struct {
```

for clarity? the display name shouldn't be used elsewhere